R/utilities-simulation.R

```diff
@@ -122,6 +122,59 @@ runSimulation <- function(simulation, population = NULL, simulationRunOptions =
   SimulationResults$new(results, simulation)
 }
 
+#' @title  Creates and returns an instance of a \code{SimulationBatch} that can be used to efficiently vary parameters and initial values in a simulation
+#'
+#' @param simulation Instance of a \code{Simulation} to simulate in a batch mode
+#' @param parametersOrPaths  Parameter instances (element or vector) typically retrieved using
+#' \code{getAllParametersMatching} or parameter path (element or vector of strings) that will be varied in the simulation. (optional)
+#' When providing the paths, only absolute full paths are supported (i.e., no matching with '*' possible).
+#' If parametersOrPaths is \code{NULL}, you will not be able to set parameter values during batch run.
+#'
+#' @param moleculeOrPaths  Molecule instances (element or vector) typically retrieved using
+#' \code{getAllMoleculesMatching} or molecule path (element or vector of strings) that will be varied in the simulation. (optional)
+#' When providing the paths, only absolute full paths are supported (i.e., no matching with '*' possible).
+#' If moleculeOrPaths is \code{NULL}, you will not be able to set molecule initial values during batch run.
+#'
+#' @return SimulationBatch that can be used to vary parameter values or molecule initial values and run simulation in an optimized manner
+#'
+#' @examples
+#' simPath <- system.file("extdata", "simple.pkml", package = "ospsuite")
+#' sim <- loadSimulation(simPath)
+#'
+#' # Create a simulation batch that will allow batch run for one parameter value
+#' simulationBatch <- createSimulationBatch(sim, "Organism|Liver|Volume")
+#'
+#' # Create a simulation batch that will allow batch run for multiple parameter values and initial values
+#' simulationBatch <- createSimulationBatch(sim, c("Organism|Liver|Volume", "R1-k1"), c("Organism|Liver|A"))
+#' @export
+createSimulationBatch <- function(simulation, parametersOrPaths = NULL, moleculesOrPaths = NULL) {

```

Comment:
@PavelBal @StephanSchaller @Yuri05 This is the entry point
One creates a simulation batch object passing a bunch of parameters (one or more as path or instances) or molecules
Then the batch will be the one object tot use to run simulations